libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -170,6 +170,20 @@ float SRV_Channel::get_output_norm(void)
     return ret;
 }
 
+float SRV_Channel::get_output_norm_from_pwm(uint16_t pwm)
+{
+    float ret;
+    if (pwm < servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_trim - servo_min);
+    } else if (pwm > servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_max  - servo_trim);
+    } else {
+        ret = 0;
+    }
+    
```

Comment:
@EShamaev 

> It does not need reverse as PWM was already reversed when it became PWM...

We can't just think in the current caller, we need to think in all potential users of this API. This gets a random PWM value (by the way, we need to constrain *ret* value to -1/1) and there can be arguments on both sides of wanting to do reverse or not. We can add a parameter to the method, but I think it should default to doing reverse.

> ESC scaling in HAL is legacy I believe, as it uses same ranges for all ESCs that might be present. This one is more agile.

It isn't legacy in any way. In fact it is used by the motors library - which doesn't use this library min/max values, meaning this won't work for Copter without a lot of trouble.

@tridge I think this is actually simple: it is just changing scale - it could be implemented in a different way and still be correct (and no trim or mid used):

```
new_-1/1_scale = (pwm - servo_min) / (servo_max - servo_min) * 2 - 1
```

This shows using trim is incorrect, it should use mid point.

Regarding ESC I also think it is simple, add a getter for ESC scaling and in the AP_UAVCAN library for the ESC messages do the math to change to change from ESC scaling to -1/1 instead of using this method.